tensorflow_probability/python/bijectors/real_nvp_test.py

```diff
@@ -97,6 +97,50 @@ def testBatchedBijectorWithMLPTransform(self):
       self.assertAllClose(x_, inverse_y_, rtol=1e-1, atol=0.)
       self.assertAllClose(ildj_, -fldj_, rtol=1e-6, atol=0.)
 
+  def testBijectorConditionKwargs(self):
+    batch_size = 3
+    x_ = np.arange(batch_size * 4 * 2).astype(np.float32).reshape(
+        batch_size, 4 * 2)
+    conditions = {
+        "a":
+        tf.reshape(
+            tf.range(batch_size * 4, dtype=tf.float32), (batch_size, 4)),
+        "b":
+        tf.reshape(
+            tf.range(batch_size * 2, dtype=tf.float32), (batch_size, 2)),
+    }
+
+    nvp = tfb.RealNVP(
+        num_masked=4, validate_args=True, **self._real_nvp_kwargs)
+    x = tf.constant(x_)
+
+    forward_x = nvp.forward(x, **conditions)
+    # Use identity to invalidate cache.
+    inverse_y = nvp.inverse(tf.identity(forward_x), **conditions)
+    forward_inverse_y = nvp.forward(inverse_y, **conditions)
+    fldj = nvp.forward_log_det_jacobian(x, event_ndims=1, **conditions)
+    # Use identity to invalidate cache.
+    ildj = nvp.inverse_log_det_jacobian(
+        tf.identity(forward_x), event_ndims=1, **conditions)
+    self.evaluate(tf.global_variables_initializer())
+    [
+        forward_x_,
+        inverse_y_,
+        forward_inverse_y_,
+        ildj_,
+        fldj_,
+    ] = self.evaluate([
+        forward_x,
+        inverse_y,
+        forward_inverse_y,
+        ildj,
+        fldj,
+    ])
+    self.assertEqual("real_nvp", nvp.name)
+    self.assertAllClose(forward_x_, forward_inverse_y_, rtol=1e-6)
```

Comment:
Actually it seems like the tests are failing on our internal system...
To unblock you, can you change these to be:

      self.assertAllClose(forward_x_, forward_inverse_y_, rtol=1e-1, atol=0.)	      
      self.assertAllClose(x_, inverse_y_, rtol=1e-1, atol=0.)
      self.assertAllClose(ildj_, -fldj_, rtol=1e-6, atol=0.)

I'm going to file a bug internally to investigate where this numerical instability comes from, and then we can fix it wholesale for these tests (and you don't have to worry about it).

Reply:
Changed.